### PR TITLE
T989: Added rawdump feature

### PR DESCRIPTION
**Pull request by supragya**
Allows dumping of Interleaved array,
Signed off by: Supragya Raj<supragyaraj@gmail.com>

### CODE REVIEW

File: Source/OCcore/CMakeLists.txt [hunk 1]
```diff
@@ -54,7 +54,7 @@ endif()
 #endforeach()
 #list(REMOVE_DUPLICATES INCLUDE_DIRS)
 
-ADD_LIBRARY(${PROJECT_NAME} SHARED ${SOURCE_FILES} ${PLATFORM_FILES})
+ADD_LIBRARY(${PROJECT_NAME} SHARED ${SOURCE_FILES} ${PLATFORM_FILES} RawDump/rawdump.h RawDump/rawdump.cpp)
```
Comment (BAndiT1983) [inline Source/OCcore/CMakeLists.txt]: Why are this files added manually, as everything should be retrieved automatically? They should be moved to some specific folder, maybe also /Image.

File: Source/OCcore/RawDump/rawdump.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    os.open(fileName, std::ios::binary);
+    if (!os.is_open())
+    {
+        std::string err = "OcRawDump: File couldn't be opened: ";
+        err.append(fileName);
+        OC_LOG_ERROR(err);
+        return false;
+    }
+    std::string begin = "OcRawDump: Begin dumping in: ";
+    begin.append(fileName);
```
Comment (BAndiT1983) [inline Source/OCcore/RawDump/rawdump.cpp]: fileName can also be added by using +, e.g. ".... be opened: " + fileName;

File: Source/OCcore/RawDump/rawdump.h [hunk 3]
```diff
@@ -0,0 +1,12 @@
+//
+// Created by supragya on 13/4/18.
+//
+
+#ifndef APERTUSOPENCINE_RAWDUMP_H
+#define APERTUSOPENCINE_RAWDUMP_H
+
+namespace OcRawDump{
```
Comment (BAndiT1983) [inline Source/OCcore/RawDump/rawdump.h]: Namespaces in OC start usually with namespace OC, then area, like Image or something like that, so it should be OC::Image::RawDump

File: Source/OCcore/RawDump/rawdump.h [hunk 4]
```diff
@@ -0,0 +1,12 @@
+//
+// Created by supragya on 13/4/18.
+//
+
+#ifndef APERTUSOPENCINE_RAWDUMP_H
```
Comment (BAndiT1983) [inline Source/OCcore/RawDump/rawdump.h]: No need to give it that long names, in my opinion, just file name is ok

File: Source/OCcore/CMakeLists.txt (hunk 1 above)
Comment (supragya) [reply Source/OCcore/CMakeLists.txt]: I am sorry for this, CLion defaults added these.

File: Source/OCcore/RawDump/rawdump.cpp (hunk 2 above)
Comment (supragya) [reply Source/OCcore/RawDump/rawdump.cpp]: corrected in #38

File: Source/OCcore/RawDump/rawdump.h (hunk 3 above)
Comment (supragya) [reply Source/OCcore/RawDump/rawdump.h]: corrected in #38

File: Source/OCcore/RawDump/rawdump.h (hunk 4 above)
Comment (supragya) [reply Source/OCcore/RawDump/rawdump.h]: corrected in #38, RAWDUMP_H
